File: office-utils/src/main/scala/io/github/ppdzm/utils/office/excel/chart/Series.scala
package io.github.ppdzm.utils.office.excel.chart

import io.github.ppdzm.utils.office.excel.sheet.OOXMLSheet
import org.apache.poi.ss.usermodel.charts.ChartDataSource

/**
 * @author Created by Stuart Alex on 2019/3/29
 */
trait Series {
    protected lazy val xAxisValues: Array[String] = ooxmlSheet.getColumnData(xAxisColumnName).distinct
    protected lazy val seriesMapping: Map[String, ChartDataSource[Number]] = getSeries
    protected val ooxmlSheet: OOXMLSheet
    protected val xAxisColumnName: String

    def getSeries: Map[String, ChartDataSource[Number]]

}

File: office-utils/src/main/java/io/github/ppdzm/utils/office/excel/chart/Series.java
package io.github.ppdzm.utils.office.excel.chart;

import org.apache.poi.ss.usermodel.charts.ChartDataSource;

import java.util.Map;

/**
 * @author Created by Stuart Alex on 2023/5/16.
 */
public interface Series {
    /**
     *
     * @return
     */
    Map<String, ChartDataSource<Number>> getSeries();
}
